client/my-sites/checkout/checkout-thank-you/index.tsx

```diff
@@ -768,169 +643,6 @@ export class CheckoutThankYou extends Component<
 			)
 		);
 	};
-
-	/**
-	 * Retrieves the component (and any corresponding data) that should be displayed according to the type of purchase
-	 * just performed by the user.
-	 *
-	 * returns an array of varying size with the component instance,
-	 * then an optional purchase object possibly followed by a domain name
-	 */
-	getComponentAndPrimaryPurchaseAndDomain = (): ComponentAndPrimaryPurchaseAndDomain => {
```

Comment:
We should also keep this function, but remove all conditions except for failed purchases. 

Congrats page for plans is handled by the `<PlanOnlyThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/05caf22fe6e33b861c937eaabf113616d9fe7896/client/my-sites/checkout/checkout-thank-you/index.tsx#L607-L613

Congrats page for Google Workspace is handled by the `<GoogleWorkspaceSetupThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/456cc1137f8a1d90ccd2778eed4d14ada2986ff2/client/my-sites/checkout/checkout-thank-you/index.tsx#L636-L639

Congrats page for Titan is handled by the `<TitanSetupThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/456cc1137f8a1d90ccd2778eed4d14ada2986ff2/client/my-sites/checkout/checkout-thank-you/index.tsx#L630-L635

Congrats page for all other purchases will be handled by the `<GenericThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/05caf22fe6e33b861c937eaabf113616d9fe7896/client/my-sites/checkout/checkout-thank-you/index.tsx#L641